app.js

```diff
@@ -53,18 +56,22 @@ app.get('/', (req, res) => res.render('pages/home'));
 
 app.get('/login', (req, res) => res.render('pages/login'));
 
-app.post('/login-with-france-connect', (req, res) => res.redirect(getAuthorizationUrlForAuthentication(req.body.eidasLevel)));
+app.post('/login-with-france-connect', (req, res) => res.redirect(getAuthorizationUrlForAuthentication(req.body.eidasLevel, req)));
 
 app.get('/login-callback', oauthLoginCallback);
 
 app.get('/logout', (req, res) => res.redirect(getLogoutUrl(req.session.idToken)));
 
 app.get('/logout-callback', oauthLogoutCallback);
 
-app.get('/data', (req, res) => res.redirect(getAuthorizationUrlForData()));
+app.get('/data', (req, res) => res.redirect(getAuthorizationUrlForData(req)));
 
 app.get('/data-callback', oauthDataCallback);
 
+app.get('/debugFC', debugDataFC);
```

Comment:
Seen IRL, these 2 routes does not have acceptable names. I propose the following changes for route names:

```lang-js
app.post('/authorize-login', (req, res) => res.redirect(getAuthorizationUrlForAuthentication(req.body.eidasLevel)));
app.get('/authorize-data', (req, res) => res.redirect(getAuthorizationUrlForData()));
app.get('/user', getUser);
app.get('/data', getData);
```